language/movie_nl/main.py

```diff
@@ -0,0 +1,366 @@
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import codecs
+import glob
+import json
+import logging
+import os
+
+from googleapiclient import discovery
+from googleapiclient.errors import HttpError
+import httplib2
+from oauth2client.client import GoogleCredentials
+import requests
+import socket
+
+def analyze_document(service, document):
+    """Analyze the document and get the distribution of sentiments and
+    the movie name."""
+    logging.info('Analyzing {}'.format(document.doc_id))
+
+    sentences, entities = document.extract_all_sentences(service)
+
+    try:
+        sentiments = [get_sentiment(service, sentence) for sentence in sentences]
+    except HttpError as e:
+        raise e
+
+    return sentiments, entities
+
+
+def get_request_body(text, syntax=True, entities=True, sentiment=True):
+    """Creates the body of the request to the language api in
+    order to get an appropriate api response."""
+    body = {
+        'document': {
+            'type': 'PLAIN_TEXT',
+            'content': text,
+        },
+        'features': {
+            'extract_syntax': syntax,
+            'extract_entities': entities,
+            'extract_document_sentiment': sentiment,
+        },
+        'encoding_type': 'UTF32'
+    }
+
+    return body
+
+def get_response_with_retry(request, tries, retry=3):
+    """Get the response using re-try"""
+    try:
+        response = request.execute()
```

Comment:
FYI you can just pass in a `num_retries` parameter to `execute`, which will do nice things like exponential backoff, and also won't retry on user error.
[reference](https://google.github.io/google-api-python-client/docs/epy/googleapiclient.http.HttpRequest-class.html)